VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -175,36 +176,41 @@ private boolean isFollowingTopics() {
         return DiscussionTopic.FOLLOWING_TOPICS.equalsIgnoreCase(discussionTopic.getName());
     }
 
-    private boolean listNeedsToBeRefreshed = true;
-
-    private void scheduleRefresh() {
-        if (isResumed()) {
-            populateThreadList(true);
-        } else {
-            listNeedsToBeRefreshed = true;
-        }
-    }
-
     public void onEventMainThread(DiscussionThreadFollowedEvent event) {
-        // TODO: Optimization: Only refresh the item if the followed post is currently listed
-        // TODO: Optimization: Only refresh the row that was updated, instead of the whole list
-        scheduleRefresh();
+        for (int i = 0; i < discussionPostsAdapter.getCount(); ++i) {
+            if (discussionPostsAdapter.getItem(i).getIdentifier().equalsIgnoreCase(event.getDiscussionThread().getIdentifier())) {
```

Comment:
We should probably implement this equality checking in the `equals()` method of `DiscussionThread`. That should also take care of performing null checks (as a sanity check).